pipeline/tests/test_regressions.py

```diff
@@ -10,3 +13,172 @@ def test_issue_0002():
+                "vocab:memberOf": {
+                    "@id": "_:002",
+                    "@type": "https://openminds.ebrains.eu/core/Organization",
+                },
+            },
+            {
+                "@type": "https://openminds.ebrains.eu/core/Affiliation",
+                "vocab:memberOf": {
+                    "@id": "_:003",
+                    "@type": "https://openminds.ebrains.eu/core/Organization",
+                },
+            },
+        ],
+    }
+    assert actual == expected
+
+    c = Collection(person, uni1, uni2)
+    output_paths = c.save(
+        "issue0007.jsonld", individual_files=False, include_empty_properties=False
+    )
+    assert output_paths == ["issue0007.jsonld"]
+
+    with open(output_paths[0]) as fp:
+        saved_data = json.load(fp)
+    os.remove("issue0007.jsonld")
+    expected_saved_data = {
+        "@context": {"vocab": "https://openminds.ebrains.eu/vocab/"},
+        "@graph": [
+            {
+                "@id": "_:001",
+                "@type": "https://openminds.ebrains.eu/core/Person",
+                "vocab:affiliation": [
+                    {
+                        "@type": "https://openminds.ebrains.eu/core/Affiliation",
+                        "vocab:memberOf": {
+                            "@id": "_:002",
+                            "@type": "https://openminds.ebrains.eu/core/Organization",
+                        },
+                    },
+                    {
+                        "@type": "https://openminds.ebrains.eu/core/Affiliation",
+                        "vocab:memberOf": {
+                            "@id": "_:003",
+                            "@type": "https://openminds.ebrains.eu/core/Organization",
+                        },
+                    },
+                ],
+                "vocab:familyName": "Professor",
+                "vocab:givenName": "A",
+            },
+            {
+                "@id": "_:002",
+                "@type": "https://openminds.ebrains.eu/core/Organization",
+                "vocab:fullName": "University of This Place",
+            },
+            {
+                "@id": "_:003",
+                "@type": "https://openminds.ebrains.eu/core/Organization",
+                "vocab:fullName": "University of That Place",
+            },
+        ],
+    }
+    assert saved_data == expected_saved_data
+
+
+def test_issue0008():
+    # https://github.com/openMetadataInitiative/openMINDS_Python/issues/8
+    # The instance of linked types in instances of embedded types are integrated as embedded not linked
+    # (example: person -> affiliation (embedded) -> organization (linked))
+
+    uni1 = omcore.Organization(full_name="University of This Place", id="_:001")
+    person = omcore.Person(
+        id="_:002",
+        given_name="A",
+        family_name="Professor",
+        affiliation=[omcore.Affiliation(member_of=uni1, end_date=date(2023, 9, 30))],
```

Comment:
the issue considering this was created by falsely entering `end_date=(2023, 9, 30)` which then was saved as a list of numbers. so this test does not capture the original issue (we should probably keep it though, since it covers the correct behavior)

Reply:
the original issue is #5, addressed in PR #19

Reply:
okay